doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,377 @@
+  - [References](#references)
+
+## Revision ##
+
+| Rev | Date       | Author           | Change Description |
+| --- | ---------- | ---------------- | ------------------ |
+| 0.1 | 05/16/2024 | Vasundhara Volam | Initial version    |
+| 0.2 | 05/29/2024 | Vasundhara Volam | Update images and APIs |
+| 0.3 | 06/10/2024 | Vasundhara Volam | Minor changes based on discussion with the community |
+| 0.4 | 07/29/2024 | Vasundhara Volam | Add PCIe daemon changes |
+
+## Glossory ##
+
+| Term  | Meaning                                   |
+| ----- | ----------------------------------------- |
+| ASIC  | Application-Specific Integrated Circuit   |
+| DPU   | Data Processing Unit                      |
+| gNMI  | gRPC Network Management Interface         |
+| gNOI  | gRPC Network Operations Interface         |
+| NPU   | Network Processing Unit                   |
+| PCI-E | Peripheral Component Interconnect Express |
+
+## Overview ##
+
+Smart Switch aims to provide a full suite of network functionality, like traditional network devices, but with the flexibility
+and scalability of cloud-based services. It consists of one switch ASIC (NPU) and multiple DPUs. The DPU ASICs are only
+connected to the NPU, and all front panel ports are connected to the NPU. The DPU also connects to the SmartSwitch CPU via PCI-E
+interfaces, allowing the Switch CPU to control DPUs through these interfaces.
+
+Each DPU will have one internal management IP which is used for internal communications, such as Redis database and zmq. This
+internal communication is also used between NPU and DPU, and between DPUs.
+
+This document provides high level design of reboot sequence of a SmartSwitch with multiple DPUs and single DPU reboot sequence
+through GNOI API.
+
+## Assumptions ##
+
+Smart Switch supports only cold-reboot and does not support warm-reboot as of today.
+
+## Requirements ##
+
+1. NPU host is running gNMI/gNOI server to communicate with DPU.
+2. DPU host is running gNOI server to listen to gNOI client requests.
+3. Each DPU is assigned an IP address to communicate from NPU.
+
+## Methods of Switch and DPU Reboot ##
+
+The switch or DPU can be rebooted using either the CLI or during an image upgrade. The reboot can be initiated in two ways.
+
+1. Performing a complete SmartSwitch reboot, which restarts the NPU and all DPUs.
+2. Rebooting a specific DPU by specifying its ID.
+
+In addition to the previously mentioned causes of graceful reboots, a switch or DPU may also reboot due to events such as power failures during DPU power-up, kernel panics, and other similar incidents.
+
+## DPU reboot sequence ##
+
+<p align="center"><img src="./images/dpu-reboot-seq.svg"></p>
+
+DPUs are internally connected to the NPU via PCI-E bridge. Below is the reboot sequence for rebooting a specific DPU:
+
+* Upon receiving a reboot CLI command to restart a particular DPU, the NPU transmits a gNOI Reboot API signal with reboot method set to ‘HALT’, instructing
```

Comment:
Can you specify which reboot CLI thing refers to?

Reply:
This is regular linux "reboot" command.